async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -1095,6 +1047,29 @@ impl futures::Stream for Stream {
         if self.terminated {
             return Poll::Ready(None);
         }
+        if !self.batch_config.idle_heartbeat.is_zero() {
+            trace!("setting hearbeats");
+            if self.heartbeat_timeout.is_none() {
+                self.heartbeat_timeout = Some(Box::pin(tokio::time::timeout(
+                    self.batch_config.idle_heartbeat.saturating_mul(2),
+                    futures::future::pending(),
+                )));
+            }
+
+            trace!("checking idle hearbeats");
+            if let Some(hearbeat) = self.heartbeat_timeout.as_mut() {
```

Comment:
This could be done via [`Option::get_or_insert`](https://doc.rust-lang.org/nightly/std/option/enum.Option.html#method.get_or_insert)

Reply:
Ah, forgot to improve that part after successful testing. Thanks!